### PR TITLE
SSCS-8407 Adding ESA Scenario Integration Tests

### PR DESCRIPTION
**Pull request by michaellavelle**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-8407

### Change description ###

Adding ESA Scenario Integration Tests

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 1]
```diff
@@ -0,0 +1,1672 @@
...
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertEquals("Something else.", payload.getAnythingElse());
+        assertNotNull(parentPayload.getWriteFinalDecisionTemplateContent());
+        EsaTemplateContent esaTemplateContent = (EsaTemplateContent)parentPayload.getWriteFinalDecisionTemplateContent();
+        Assert.assertEquals(EsaScenario.SCENARIO_4, esaTemplateContent.getScenario());
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario4_WhenIncorrectlyRefused() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:681]: willNotPreview

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 2]
```diff
@@ -0,0 +1,1672 @@
...
+
+        assertEquals(1, result.getErrors().size());
+
+        // This combination is not possible with the correctly configured ccd page flow - just assert that we defensively prevent document generation
+        assertEquals("http://dm-store:5005/documents/7539160a-b124-4539-b7c1-f3dcfbcea94c", result.getData().getWriteFinalDecisionPreviewDocument().getDocumentUrl());
+        verifyNoInteractions(generateFile);
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willNotPreviewTheDocumentForScenario5() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:728]: willPreviewTheDocument

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 3]
```diff
@@ -0,0 +1,1672 @@
...
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertEquals("Something else.", payload.getAnythingElse());
+        assertNotNull(parentPayload.getWriteFinalDecisionTemplateContent());
+        EsaTemplateContent esaTemplateContent = (EsaTemplateContent)parentPayload.getWriteFinalDecisionTemplateContent();
+        Assert.assertEquals(EsaScenario.SCENARIO_6, esaTemplateContent.getScenario());
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario6_WhenIncorrectlyRefused() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:919]: will**Not**PerviewTheDocument

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 4]
```diff
@@ -0,0 +1,1672 @@
...
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertEquals("Something else.", payload.getAnythingElse());
+        assertNotNull(parentPayload.getWriteFinalDecisionTemplateContent());
+        EsaTemplateContent esaTemplateContent = (EsaTemplateContent)parentPayload.getWriteFinalDecisionTemplateContent();
+        Assert.assertEquals(EsaScenario.SCENARIO_7, esaTemplateContent.getScenario());
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario7_WhenIncorrectlyRefused() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:1077]: will**Not**PreviewTheDocument

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 5]
```diff
@@ -0,0 +1,1672 @@
...
+
+        assertEquals("http://dm-store:5005/documents/7539160a-b124-4539-b7c1-f3dcfbcea94c", result.getData().getWriteFinalDecisionPreviewDocument().getDocumentUrl());
+
+        // This combination is not possible with the correctly configured ccd page flow - just assert that we defensively prevent document generation
+        assertEquals("http://dm-store:5005/documents/7539160a-b124-4539-b7c1-f3dcfbcea94c", result.getData().getWriteFinalDecisionPreviewDocument().getDocumentUrl());
+        verifyNoInteractions(generateFile);
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario7_WhenSchedule3NotSet() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:1125]: will**Not**Preview

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 6]
```diff
@@ -0,0 +1,1672 @@
...
+
+        assertEquals("http://dm-store:5005/documents/7539160a-b124-4539-b7c1-f3dcfbcea94c", result.getData().getWriteFinalDecisionPreviewDocument().getDocumentUrl());
+
+        // This combination is not possible with the correctly configured ccd page flow - just assert that we defensively prevent document generation
+        assertEquals("http://dm-store:5005/documents/7539160a-b124-4539-b7c1-f3dcfbcea94c", result.getData().getWriteFinalDecisionPreviewDocument().getDocumentUrl());
+        verifyNoInteractions(generateFile);
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario7_WhenRegulation29IncorrectlyDoesNotApply() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:1150]: willNotPreviewTheDocument

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 7]
```diff
@@ -0,0 +1,1672 @@
...
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertEquals("Something else.", payload.getAnythingElse());
+        assertNotNull(parentPayload.getWriteFinalDecisionTemplateContent());
+        EsaTemplateContent esaTemplateContent = (EsaTemplateContent)parentPayload.getWriteFinalDecisionTemplateContent();
+        Assert.assertEquals(EsaScenario.SCENARIO_8, esaTemplateContent.getScenario());
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario8_WhenIncorrectlyRefused() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:1237]: willNotPreviewTheDocument

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 8]
```diff
@@ -0,0 +1,1672 @@
...
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertEquals("Something else.", payload.getAnythingElse());
+        assertNotNull(parentPayload.getWriteFinalDecisionTemplateContent());
+        EsaTemplateContent esaTemplateContent = (EsaTemplateContent)parentPayload.getWriteFinalDecisionTemplateContent();
+        Assert.assertEquals(EsaScenario.SCENARIO_9, esaTemplateContent.getScenario());
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario9_WhenIncorrectlyRefused() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:1320]: will**Not**Preview

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java [hunk 9]
```diff
@@ -0,0 +1,1672 @@
...
+        assertEquals(1, payload.getReasonsForDecision().size());
+        Assert.assertEquals("My reasons for decision", payload.getReasonsForDecision().get(0));
+        assertEquals("Something else.", payload.getAnythingElse());
+        assertNotNull(parentPayload.getWriteFinalDecisionTemplateContent());
+        EsaTemplateContent esaTemplateContent = (EsaTemplateContent)parentPayload.getWriteFinalDecisionTemplateContent();
+        Assert.assertEquals(EsaScenario.SCENARIO_12, esaTemplateContent.getScenario());
+    }
+
+    @Test
+    public void callToMidEventPreviewFinalDecisionCallback_willPreviewTheDocumentForScenario12_WhenIncorrectlyRefused() throws Exception {
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/EsaWriteFinalDecisionIt.java:1600]: will**Not**PreviewTheDocument

File: src/main/resources/config/application.properties [hunk 10]
```diff
@@ -37,17 +37,17 @@ core_case_data.jurisdictionId=${CORE_CASE_DATA_JURISDICTION_ID:SSCS}
 core_case_data.caseTypeId=${CORE_CASE_DATA_CASE_TYPE_ID:Benefit}
 
 # IDAM Java Client
-idam.url=${IDAM_API_URL:http://localhost:5000}
+idam.url=${IDAM_API_URL:http://localhost:5555}
```
Comment (ryandac) [inline src/main/resources/config/application.properties:40]: Is it okay to change to port 5555?
Some devs, (cough! me!) are using the real idam and not the stub.

**Review by ryandac — CHANGES_REQUESTED** [review]

File: src/main/resources/config/application.properties (hunk 10 above)
Comment (michaellavelle) [reply src/main/resources/config/application.properties]: Good catch Ryan!  Mistaken checkin - thanks for spotting :-)

**Review by ryandac — APPROVED** [review]

**Pull request merged**